### PR TITLE
Create shipments (closes: #18 #27 #28)

### PR DESCRIPTION
**Pull request by jritsema**
- implemented [shipment/environment creation in`up`](https://github.com/turnerlabs/harbor-compose/issues/18)
- implemented [`init` command](https://github.com/turnerlabs/harbor-compose/issues/28)
- implemented [`down --delete`](https://github.com/turnerlabs/harbor-compose/issues/27)

### CODE REVIEW

File: cmd/harborAPI.go [hunk 1]
```diff
@@ -305,3 +325,43 @@ func UpdateContainerImage(token string, shipment string, composeShipment Compose
...
+
+	if err != nil || res.StatusCode != 200 {
+		fmt.Printf("creating shipment was not successful: %v \n", body)
+		return false
+	}
+
+	//api returns an object with an errors property that is
+	//false when there are no errors and an object if there are
+	if !strings.Contains(body, "errors\": false") {
+		return false
```
Comment (jkurz) [inline cmd/harborAPI.go:343]: should you log the errors for the user?

File: cmd/init.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	group := "mss"
+	property := "turner.com"
+	project := "poc"
+	product := "poc"
+
+	//ask questions
+	if !yesUseDefaults {
+		name = promptAndGetResponse("shipment name: (e.g., mss-my-app) ")
+		env = promptAndGetResponse("shipment environment: (dev, qa, prod, etc.) ")
+		barge = promptAndGetResponse("barge: (cnn, nba, digital-sandbox, corp-sandbox, etc.) ")
```
Comment (jkurz) [inline cmd/init.go]: can you actually list all the available options here for now. At least the user will get some insight into what they can choose.

File: cmd/init.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		Env:      env,
+		Group:    group,
+		Property: property,
+		Project:  project,
+		Product:  product,
+	}
+
+	composeShipment.Barge = barge
+	intReplicas, err := strconv.Atoi(replicas)
+	if err != nil {
```
Comment (jkurz) [inline cmd/init.go:67]: do you want to make sure replicas is positive or zero also?

Comment (jkurz) [inline cmd/init.go:67]: nevermind, shipitapi is already doing this for us. https://github.com/turnerlabs/shipit-api/blob/master/models/helpers.js#L99

File: cmd/up.go [hunk 4]
```diff
@@ -12,8 +15,15 @@ import (
 var upCmd = &cobra.Command{
 	Use:   "up",
 	Short: "Start your application",
-	Long:  ``,
-	Run:   up,
+	Long: `The up command applies changes from your docker/harbor compose files and brings your application up on Harbor.  The up command:
```
Comment (jkurz) [inline cmd/up.go:18]: have extra "the up command:" at the end of this line.

Comment (jkurz) [inline cmd/up.go:18]: nevermind this looks good.

File: cmd/up.go [hunk 5]
```diff
@@ -52,80 +63,254 @@ func up(cmd *cobra.Command, args []string) {
...
 
-			//lookup the container in the list of services in the docker-compose file
-			dockerService := dockerCompose.Services[container]
+	//add environment-level env vars
+	for name, value := range shipment.Environment {
+		newShipment.Environment.Vars = append(newShipment.Environment.Vars, envVar(name, value))
+	}
+
+	//add barge envvar (note that this is being moved to the provider level soon)
+	if shipment.Barge == "" {
```
Comment (jkurz) [inline cmd/up.go]: you can go ahead and remove this.

File: cmd/harborAPI.go (hunk 1 above)
Comment (jritsema) [inline cmd/harborAPI.go:343]: Yes, probably a good idea.

File: cmd/init.go (hunk 2 above)
Comment (jritsema) [inline cmd/init.go]: Sure.  You mean for barges?  Just didn't want to make the lines to long ;)

File: cmd/up.go (hunk 5 above)
Comment (jritsema) [inline cmd/up.go]: Well when are you guys pushing that change?  Does it hurt anything being there for now?

File: cmd/init.go (hunk 2 above)
Comment (jkurz) [inline cmd/init.go]: yeah or maybe we create a new function later that lists available barges

File: cmd/up.go (hunk 5 above)
Comment (jkurz) [inline cmd/up.go]: it was pushed today. So now trigger reads from the provider barge value. Plus shipit won't allow you to create a new shipment without a barge value on the provider.

Comment (jritsema) [inline cmd/up.go]: I'm already setting in on the provider in addition to the old env var.  But i'll go ahead and remove the env var if it's ready to go.

File: cmd/harborAPI.go (hunk 1 above)
Comment (jritsema) [inline cmd/harborAPI.go:343]: So after looking at this more closely, if there's an error, it's currently writing the HTTP response body to the console.  I'm not entirely sure the shape and meaning of the error data coming back so I'm thinking we just leave as-is for now, until we move to the new server API.

Comment (jkurz) [inline cmd/harborAPI.go:343]: sounds good to me.

**Pull request merged**